scripts/dependency-graph-generator/index.js

```diff
@@ -0,0 +1,176 @@
+const graphviz = require('graphviz');
+const parser = require('dotparser');
+const readFileSync = require('fs').readFileSync;
+const spawnSync = require('child_process').spawnSync;
+const findGitRoot = require('../monorepo/index').findGitRoot;
+const path = require('path');
+
+const dotFilePath = path.resolve(__dirname, 'repo-graph.dot');
+const ignoreDevDependencies = [
+  '@fluentui/set-version',
+  '@fluentui/react-conformance',
+  '@fluentui/test-utilities',
+  '@fluentui/eslint-plugin',
+  '@fluentui/a11y-testing',
+  '@fluentui/jest-serializer-merge-styles',
+  '@fluentui/scripts',
+];
```

Comment:
It will be nice to collect these deps instead of hard-coding... I tried to reuse `lerna`'s `PackageGraph`, but there is no obvious way to ask only for `devDependencies`, so I have some ugly code:

```tsx
const Project = require('@lerna/project');
const PackageGraph = require('@lerna/package-graph');
const config = require('@fluentui/scripts/config.ts').default;

function flattenPackageGraph(rootPackages, projectGraph, packageList = []) {
  rootPackages.forEach(packageName => {
    packageList.push(packageName);

    flattenPackageGraph([...projectGraph.get(packageName).localDependencies.keys()], projectGraph, packageList);
  });

  return packageList.sort().filter((v, i, a) => a.indexOf(v) === i);
}

async function getDevDependencies(packageName) {
  const lernaProject = new Project(config.paths.base('packages'));
  const projectPackages = await lernaProject.getPackages();

  const allDepsGraph = flattenPackageGraph([packageName], new PackageGraph(projectPackages));
  const productionDepsGraph = flattenPackageGraph([packageName], new PackageGraph(projectPackages, 'dependencies'));

  const devDependencies = allDepsGraph.filter(dep => !productionDepsGraph.includes(dep));

  return devDependencies;
}

getDevDependencies('@fluentui/react-avatar').then(res => console.log(res));
```

Reply:
That's a fair point, I added a modified version of this that doesn't depend on config.ts